react-context/client/components/navbar.jsx

```diff
@@ -0,0 +1,37 @@
+import React from 'react';
+import AppContext from '../lib/app-context';
+export default class Navbar extends React.Component {
+  render() {
+    const { user, handleSignOut } = this.context;
+    return (
+      < AppContext.Consumer>
+        <nav className="navbar navbar-dark bg-dark">
+          <div className="container">
+            <a className="navbar-brand" href="#">
+              <i className="fas fa-bolt me-2" />
+              Awesome App
+            </a>
+            <div>
+              { user !== null &&
+                <button className="btn btn-dark" onClick={handleSignOut}>
+                  Sign out
+                  <i className="ms-2 fas fa-sign-out-alt" />
+                </button>
+              }
+              { user === null &&
+                <>
+                  <a href="#sign-in" className="btn btn-primary">
+                    Sign In
+                  </a>
+                  <a href="#sign-up" className="btn btn-dark">
+                    Sign Up
+                  </a>
+                </>
+              }
+            </div>
+          </div>
+        </nav>
+      </AppContext.Consumer>
+    );
+  }
+}
```

Comment:
After the class is defined we want to add the context here after the component definition.

```JSX
Navbar.contextType = AppContext;
```

Adding a static property after the component definition will make context available to the entire component which is what we want.